main.cpp

```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+#include <fstream>
+#include <string.h>
+
+
+using namespace std;
+
+int main(int argc, char *argv[]){
+
+    string file_path;
+    if (argv[1]){
+        file_path = static_cast<string>(argv[1]);
+
+    }else if (static_cast<string>(argv[2]) == "search"){
```

Comment:
I think the "else" may need to be removed from this line. If line one is evaluated to true, then every following "else if" won't execute. Everything looks good so far :)